src/wp-includes/l10n.php

```diff
@@ -336,6 +336,59 @@ function esc_html_e( $text, $domain = 'default' ) {
 	echo esc_html( translate( $text, $domain ) );
 }
 
+/**
+ * Retrieve the translation of $text and escapes it for safe use in XML output.
+ *
+ * If there is no translation, or the text domain isn't loaded, the original text
+ * is escaped and returned.
+ *
+ * @since 5.5.0
+ *
+ * @param string $text   Text to translate.
+ * @param string $domain Optional. Text domain. Unique identifier for retrieving translated strings.
+ *                       Default 'default'.
+ * @return string Translated text.
+ */
+function esc_xml__( $text, $domain = 'default' ) {
+	return esc_xml( translate( $text, $domain ) );
+}
+
+/**
+ * Display translated text that has been escaped for safe use in XML output.
+ *
+ * If there is no translation, or the text domain isn't loaded, the original text
+ * is escaped and displayed.
+ *
+ * If you need the value for use in PHP, use esc_xml__().
+ *
+ * @since 5.5.0
+ *
+ * @param string $text   Text to translate.
+ * @param string $domain Optional. Text domain. Unique identifier for retrieving translated strings.
+ *                       Default 'default'.
+ */
+function esc_xml_e( $text, $domain = 'default' ) {
+	echo esc_xml( translate( $text, $domain ) );
+}
+
+/**
+ * Translate string with gettext context, and escapes it for safe use in XML output.
+ *
+ * If there is no translation, or the text domain isn't loaded, the original text
+ * is escaped and returned.
+ *
+ * @since 5.5.0
+ *
+ * @param string $text    Text to translate.
+ * @param string $context Context information for the translators.
+ * @param string $domain  Optional. Text domain. Unique identifier for retrieving translated strings.
+ *                        Default 'default'.
+ * @return string Translated text.
+ */
+function esc_xml_x( $text, $context, $domain = 'default' ) {
+	return esc_xml( translate_with_gettext_context( $text, $context, $domain ) );
+}
```

Comment:
These i18n helpers seem to be out-of-scope and I'm rather against adding them. They aren't used at all in this PR? Adding them would also require updating our tooling. `esc_xml( __() )` should be fine enough for the moment.

Reply:
This PR is still a WIP, but they would be eventually used. But forgot about the tooling implications. Happy to defer those to a separate PR 👍

Reply:
Removed for now.